tlsr9/CMakeLists.txt

```diff
@@ -70,50 +89,88 @@ if(CONFIG_BT_B9X OR CONFIG_IEEE802154_TELINK_B9X)
 		common/b9x_rf_power.c
 	)
 endif()
+if(CONFIG_BT_TLX OR CONFIG_IEEE802154_TELINK_TLX)
+	zephyr_library_sources(
+		drivers/${SOC}/aes.c
+		common/b9x_rf_power.c
+	)
+endif()
 
 # ADC driver reference sources
 zephyr_library_sources_ifdef(CONFIG_ADC_TELINK_B9X drivers/${SOC}/adc.c)
 zephyr_library_sources_ifdef(CONFIG_ADC_TELINK_B9X drivers/${SOC}/gpio.c)
+zephyr_library_sources_ifdef(CONFIG_ADC_TELINK_TLX drivers/${SOC}/adc.c)
+zephyr_library_sources_ifdef(CONFIG_ADC_TELINK_TLX drivers/${SOC}/gpio.c)
 
 # USB driver reference sources
 zephyr_library_sources_ifdef(CONFIG_USB_TELINK_B9X drivers/${SOC}/gpio.c)
 zephyr_library_sources_ifdef(CONFIG_USB_TELINK_B9X drivers/${SOC}/usbhw.c)
+zephyr_library_sources_ifdef(CONFIG_USB_TELINK_TLX drivers/${SOC}/gpio.c)
+zephyr_library_sources_ifdef(CONFIG_USB_TELINK_TLX drivers/${SOC}/usbhw.c)
 
 #PM driver dependency
+if (NOT CONFIG_SOC_RISCV_TELINK_TL321X)
 zephyr_library_sources_ifdef(CONFIG_PM drivers/${SOC}/stimer.c)
+endif()
 zephyr_library_sources_ifdef(CONFIG_PM common/sleep.c)
 
 #BLE flash reference sources
-if (CONFIG_IEEE802154_TELINK_B9X)
+if (CONFIG_IEEE802154_TELINK_B9X OR CONFIG_IEEE802154_TELINK_TLX)
 
-zephyr_include_directories(
+zephyr_include_directories_ifdef(CONFIG_IEEE802154_TELINK_B9X
 	ble
-	ble/vendor/controller
+	ble/vendor/controller/b9x
+)
+
+zephyr_include_directories_ifdef(CONFIG_IEEE802154_TELINK_TLX
+	ble
+	ble/vendor/controller/tlx
 )
 
 endif()
 
 # BLE stack reference sources
-if (CONFIG_BT_B9X)
+if (CONFIG_BT_B9X OR CONFIG_BT_TLX)
+
+zephyr_include_directories_ifdef(CONFIG_BT_B9X
+	ble
+	ble/vendor/controller/b9x
+)
 
-zephyr_include_directories(
+zephyr_include_directories_ifdef(CONFIG_BT_TLX
 	ble
-	ble/vendor/controller
+	ble/vendor/controller/tlx
 )
 
-zephyr_library_sources(
+zephyr_library_sources_ifdef(CONFIG_BT_B9X
 	ble/common/utility.c
-	ble/vendor/controller/b9x_bt.c
-	ble/vendor/controller/b9x_bt_init.c
-	ble/vendor/controller/b9x_bt_buffer.c
-	ble/vendor/controller/b9x_bt_flash.c
+	ble/vendor/controller/b9x/b9x_bt.c
+	ble/vendor/controller/b9x/b9x_bt_init.c
+	ble/vendor/controller/b9x/b9x_bt_buffer.c
+	ble/vendor/controller/b9x/b9x_bt_flash.c
 )
 
-endif() # CONFIG_BT_B9X
+zephyr_library_sources_ifdef(CONFIG_BT_TLX
+	ble/common/utility.c
+	ble/vendor/controller/tlx/tlx_bt.c
+	ble/vendor/controller/tlx/tlx_bt_init.c
+	ble/vendor/controller/tlx/tlx_bt_buffer.c
+	ble/vendor/controller/tlx/tlx_bt_flash.c
+)
```

Comment:
Here make sense to rename ble controller folders same as for drivers to use ${SOC} and simplify code
For files "b9x_bt*, tlx_bt*" you can use ${SOC} as well, but need to rename files or just remove b9x_ and tlx_ from files name to have more simple code as well. Its up to you.

```
if (CONFIG_BT_B9X OR CONFIG_BT_TLX)

zephyr_include_directories(
	ble
	ble/vendor/controller/${SOC}
)

zephyr_library_sources(
	ble/common/utility.c
	ble/vendor/controller/${SOC}/bt.c
	ble/vendor/controller/${SOC}/bt_init.c
	ble/vendor/controller/${SOC}/bt_buffer.c
	ble/vendor/controller/${SOC}/bt_flash.c
)

endif() # (CONFIG_BT_B9X OR CONFIG_BT_TLX)
```

Reply:
${SOC} are `B91`, `B92`, `B95`, `TL321X`.
However, there are only two sub-folders `b9x` and `tlx` under the `controller` folder.
I wonder if I can add
`set(SOC "TL321X")` at the beginning of the file,
so I can include files from 
`ble/vendor/controller/b9x/b9x_bt.c` or `ble/vendor/controller/tlx/tlx_bt.c`
to
`ble/vendor/controller/${SOC_SERIES}/${SOC_SERIES}_bt.c`

Reply:
ok this is up to you with some optimizations

Reply:
updated using `string(REGEX REPLACE "^telink_" "" SOC_FAMILY ${CONFIG_SOC_SERIES})`